src/visualizers-workbench/P5VisualizerTemplate.ts

```diff
@@ -0,0 +1,174 @@
+import {P5Visualizer} from '../visualizers/P5Visualizer'
+import {VisualizerExportModule} from '@/visualizers/VisualizerInterface'
+
+/** md
+# p5 Visualizer Template
+
+[image should go here]
+
+This template can be used as a starting point for designing a p5 visualizer. You
+can use it to step back and forth through a sequence, showing each element as a
+written number.
+
+## Parameters
+
+- Step size
+
+## Interactions
+
+- Use the left and right arrow keys to step
+**/
+
+class P5VisualizerTemplate extends P5Visualizer {
+    // the name that appears in bundle card titles
+    static visualizationName = 'p5 Visualizer Template'
+
+    // external parameters, which the user can choose while creating the
+    // visualizer bundle
+    stepSize = 1 // the default value of this parameter is zero
+    params = {
+        /** md
+- stepSize: How far to step when the user presses a left or right arrow key.
+         **/
+        stepSize: {
+            value: this.stepSize,
+            forceType: 'integer',
+            displayName: 'Step size',
+            required: true,
+        },
+    }
+
+    // internal parameters. the value of `index` will always be overwritten in
+    // `inhabit()`, but TypeScript can't infer that, so we have to give `index`
+    // an initial value
+    index = 0
+    justStepped = false
+    flash = 0
+
+    checkParameters() {
+        const status = super.checkParameters()
+
+        // make sure the step size is positive
+        if (this.params.stepSize.value <= 0) {
+            status.isValid = false
+            status.errors.push('Step size must be positive')
+        }
+
+        return status
+    }
+
+    inhabit(element: HTMLElement): void {
+        // do the behind-the-scenes work of managing p5 canvases. this work is
+        // finicky, and your visualizer won't work unless it's done properly, so
+        // you should leave it to the superclass unless you have a very
+        // compelling reason to do it yourself
+        super.inhabit(element)
+    }
+
+    setup() {
+        // the base setup function includes the `createCanvas()` call that must
+        // appear in every p5 setup function
+        //
+        //   https://p5js.org/reference/#/p5/createCanvas
+        //
+        super.setup()
+
+        // accessing `this.sketch` triggers some consistency checks, so it's
+        // most parsimonious to access it at most once per function call,
+        // storing its value as a local constant for future reference
+        const sketch = this.sketch
+
+        // set the stroke color and text alignment, which won't change from
+        // frame to frame
+        sketch.stroke(51, 51, 255)
+        sketch.textAlign(sketch.CENTER, sketch.CENTER)
+
+        // start at the beginning of the sequence
+        this.index = this.seq.first
+    }
+
+    draw() {
```

Comment:
might want a comment just before here noting this is the core of any visualizer, and it is mandatory to implement. Up to you.

Reply:
Please just confirm you decided not to do this.

Reply:
Done.